### ISSUE TITLE
Forward Function of Metric not working after jit scripting

### ISSUE THREAD

**Issue by AdlyTempleton** [PyTorchLightning/metrics#442]
Hello! I am trying to create new torchscriptable metrics extending torchmetrics.Metric. I believe it should be possible to do this; I see unit tests which test scriptability (ie, line 297 of https://github.com/.../21fe0ca.../tests/bases/test_metric.py). I can perform the scripting, however, I get the following error when trying to call the forward function of a scripted Metric:
'RecursiveScriptModule' object has no attribute '_defaults'

Minimal Reproducing Example:

import torch
from torchmetrics import Metric
class MyMetric(Metric):
    @torch.jit.export
    def update(self, x):
        return x
        
    @torch.jit.export
    def compute(self, x):
        return x

metric = torch.jit.script(MyMetric())
metric(torch.Tensor([1]))

Is this supported behavior?

**Comment by ananthsub** [comment]
@AdlyTempleton - metrics weren't designed to be torchscripted. See here: https://github.com/PyTorchLightning/metrics/blob/caf7b0bc46f93641316af88256d838de3516f134/torchmetrics/metric.py#L176-L177

where the forward is explicitly marked as unused

**Comment by SkafteNicki** [comment]
@ananthsub ideally uses should have the option to make their metrics scriptable. I definitly think that it is something that we should have in mind for the re-design of the metrics if we can make them scriptable in some way.
